src/sage/combinat/symmetric_group_algebra.py

```diff
@@ -1941,6 +1943,42 @@ def _dft_seminormal(self, mult='l2r'):
         snb = self.seminormal_basis(mult=mult)
         return matrix([vector(b) for b in snb]).inverse().transpose()
 
+    def _dft_modular(self):
+        """
+        Return the discrete Foruier transform when the characterisc divides the order of the group.
+        The usual .dft() throws ZeroDivisionError when p|n.
+
+        EXAMPLES::
+
+            sage: GF2S3 = SymmetricGroupAlgebra(GF(2),3)
+            sage: GF2S3._dft_modular()
+            [1 0 0 0 1 0]
+            [0 1 0 0 0 1]
+            [0 0 1 0 0 1]
+            [0 0 0 1 1 0]
+            [1 0 0 1 1 0]
+            [0 1 1 0 0 1]
```

Comment:
This example should go into the `.dft`-method, so that it shows up in the documentation, with the small change given below:

```suggestion
            sage: GF2S3 = SymmetricGroupAlgebra(GF(2),3)
            sage: GF2S3.dft(form="modular")
            [1 0 0 0 1 0]
            [0 1 0 0 0 1]
            [0 0 1 0 0 1]
            [0 0 0 1 1 0]
            [1 0 0 1 1 0]
            [0 1 1 0 0 1]
```